### ISSUE TITLE
branch method does not respect existing node aliases

### ISSUE THREAD

**Issue by klobuczek** [neo4jrb/neo4j#1348]
`app_role.includes.branch { privileges }` generates incorrect alias in the query (result_includes)

while explicit aliasing works properly: `app_role.includes(:i).branch { privileges }`

Additional information which could be helpful if relevant to your issue:

### Code example (inline, gist, or repo)

```
[11] pry(main)> >> app_role.includes.branch { privileges }
AppRole#includes#privileges 
MATCH (approle18) 
WHERE (ID(approle18) = {ID_approle18}) 
MATCH (approle18)-[rel1:`include`]->(node3:`AppRole`) 
MATCH (node3)-[rel2:`privilege`]->(result_privileges:`Privilege`) 
RETURN result_includes | {:ID_approle18=>18}
HTTP REQUEST: 20ms POST http://localhost:7474/db/data/transaction/commit (1 bytes)```



### Runtime information:

Neo4j database version:
`neo4j` gem version: 
`neo4j-core` gem version:

**Comment by cheerfulstoic** [comment]
I made PR #1402 which is going through CI which should address this

**Comment by cheerfulstoic** [comment]
If it passes in the morning I'll release it as a patch

**Comment by cheerfulstoic** [comment]
Released as 8.1.4
